### ISSUE TITLE
table don´t sort by date

### ISSUE THREAD

**Issue by AndyIggi** [FyiurAmron/sortablejs#47]
Hello,

we have a medium sized table  (8 colums, 44 rows). In the column 7 are different dates inserted in format dd.mm.yyyy 
The syntax  <sortable7 7=date> </sortable> is inserted .
Problem: I have tested and found out that, as soon as the table is bigger then 3 rows the sortable function by date is not working anymore. 
![sortable_bug](https://github.com/FyiurAmron/sortablejs/assets/134625530/e8b95c9e-e2bf-47ba-894b-69ef2392c371)

Could you check/help me in this case?

**Comment by FyiurAmron** [comment]
@AndyIggi can you provide a complete snippet that reproduces this? It would speed things for me.

**Comment by AndyIggi** [comment]
Hi,

you mean this?
![sortable bug 2](https://github.com/FyiurAmron/sortablejs/assets/134625530/8ad8a450-e4d5-48ae-abb7-1807f625d9c1)

**Comment by FyiurAmron** [comment]
@AndyIggi yes, but as a code block, not a pasted image :D

**Comment by AndyIggi** [comment]
<sortable8 8=date>

^Col1^Col2^Col3^Col4^Col5^Col6^Col7^Col8^Col9|
|test1|xyz| | | | | |22.01.2022| |
|Test2|abc| | | | | |19.01.2023| |
|Test3|def| | | | | |24.01.2024| |
|Test4|ghi| | | | | |13.02.2021| |

</sortable>

**Comment by AndyIggi** [comment]
Hi FyiurAmron, do you have any news regarding the Problem?

**Comment by FyiurAmron** [comment]
@AndyIggi sorry, I'm basically swamped with work ATM. I'll do take a look at it as soon as I'm able to, though, probably in the couple of coming weeks. Any debugging help is welcome, though.

**Comment by Chris75forumname** [comment]
Hi, I ran into the same problem today. This is just a hack for the date format `dd.mm.yyyy` !
Just replace the `sort_date` function in `/dokuwiki/lib/plugins/sortablejs/script.js`, like so:
```
    sort_date: function ( a, b ) {
        var a_arr = a[0].split('.');
        var a_ami = a_arr[1] + '/' + a_arr[0] + '/' + a_arr[2];

        var b_arr = b[0].split('.');
        var b_ami = b_arr[1] + '/' + b_arr[0] + '/' + b_arr[2];

        var aa = new Date( a_ami ); 
        var bb = new Date( b_ami );

    //  var aa = new Date( a[0] ); 
    //  var bb = new Date( b[0] );

        return ( aa > bb ) - ( aa < bb );
    },
```

Tested and works with the following table syntax:
```
<sortable 1=nosort 2=nosort 3=date>
^Col1^Col2^Col3^
|Test1|xyz|22.01.2022|
|Test2|ghi|08.02.2021|
|Test2|def|24.01.2024|
|test3|xyz|22.07.2025|
|Test4|ghi|13.02.2021|
|Test6|def|24.01.2024|
|test7|xyz|22.04.2020|
|Test8|ghi|13.08.2021|
|Test9|def|24.12.2024|
</sortable>
```
Cheers!

**Comment by Chris75forumname** [comment]
@FyiurAmron 
No action needed, afterall! Everything works alright. I updated the documentation to reflect the code in the `sort_date` function, which only sorts dates with american format `mm/dd/yyyy`.
Cheers!

**Comment by FyiurAmron** [comment]
@Chris75forumname well, you're not wrong, but you're not right. The current sorting is done via `new Date()` c-tor, which, in theory, should work like this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#date_time_string_format

in practice, it _usually_ accepts legacy formats (based on browser etc.), such as:
* MM-dd-yyyy
* yyyy/MM/dd
* MM/dd/yyyy
* MMMM dd, yyyy
* MMM dd, yyyy

IMVHO this should be still fixed here in the plugin to allow passing the formatting string for dates and to use locale-based default format for this, but the exact implementation of the fix is TBD.

**Comment by Chris75forumname** [comment]
@FyiurAmron 
Thank you very much for your reply and more information! I updated the plugin homepage accordingly. I think this will help alot!
Cheers!

**Comment by LMS235** [comment]
We have now also implemented the "workaround", so it now also works with DD.MM.YYYY. Can the plugin generally be extended to use the EU spelling (DD.MM.YYYY)?

**Comment by FyiurAmron** [comment]
@LMS235 yes, and it shouldn't be too difficult TBH, but the key issues are twofold: 1. my time limits, 2. my strong preference to do a proper solution via https://tc39.es/proposal-temporal/docs/index.html , which sadly isn't supported in browsers yet (see https://caniuse.com/temporal ), since it's TC39 Stage 3 now. TBH, if someone does a PR with a proper temporary fix, I'd be happy to merge it, but otherwise I think this will stay in freezer for some months till Temporal gets stable.

**Comment by FyiurAmron** [comment]
@LMS235 could you test if https://github.com/FyiurAmron/sortablejs/pull/49 works properly? I don't have time for extensive testing here, RL is eating me up completely and it's (un)fortunately unlikely to change :)

**Comment by LMS235** [comment]
@FyiurAmron I just tested it briefly with a few tables, and it seems to work.

**Comment by FyiurAmron** [comment]
@Cyrille37 can you verify the fix is published to central plugin registry and working?

**Comment by LMS235** [comment]
i've updated https://www.dokuwiki.org/plugin:sortablejs
